whatnext/routes.py

```diff
@@ -2,31 +2,31 @@
 This file contains all the endpoints for the 'What next' application
 """
 
-from flask import render_template, url_for, flash, redirect, jsonify, request, Response, session
-from sqlalchemy import or_
+from flask import render_template, jsonify, request
 from whatnext import app
-from whatnext import db
-from whatnext.models import Raw, Tagpair
+from whatnext.models import Tags
 
 @app.route("/",methods=["GET","POST"])
 def home():
     "Return the index page"
     if request.method == 'GET':
         return render_template("index.html")
+
     if request.method == 'POST':
+        tags_object = Tags()
         query_string = request.form.get('terms')
         technologies = query_string.split(',')
         associated_tech_graph = {'nodes':[],'edges':[]}
-        tech_id = 1
-        edge_id = 100
+
         for tech in technologies:
             tech = tech.strip().lower()
-            tech_pair_rows = Tagpair.query.filter(or_(Tagpair.tag1==tech, Tagpair.tag2==tech)).order_by(Tagpair.count.desc()).limit(6)
+            tech_pair_rows = tags_object.fetch_nodes(tech)
```

Comment:
If we give an empty space or some random string, its giving the same, unrelated results. I think we should validate for non empty strings. And not sure about the unrelated graph thats coming for random strings. Can you please take a look at that?

Reply:
Sravati,I have validated the empty string and some random string. But I couldnt validate ',' since I have to give input with comma separated values. So if we give only ',' it gives random graph and I beleive I have handled all other cases.

Reply:
Should be fine then, Annapoorani.